pkgs/by-name/co/comfyui/package.nix

```diff
@@ -0,0 +1,164 @@
+      vae_approx = "${modelsPath}/vae_approx";
+    };
+  };
+
+  modelPathsFile = writeTextFile {
+    name = "extra_model_paths.yaml";
+    text = (lib.generators.toYAML { } config-data);
+  };
+
+  pythonEnv = (
+    python312.withPackages (
+      ps:
+      with ps;
+      [
+        comfyui-frontend-package
+        comfyui-workflow-templates
+        torch
+        torchsde
+        torchvision
+        torchaudio
+        numpy
+        einops
+        transformers
+        tokenizers
+        sentencepiece
+        safetensors
+        aiohttp
+        yarl
+        pyyaml
+        pillow
+        scipy
+        tqdm
+        psutil
+
+        kornia
+        spandrel
+        av
+        pydantic
+      ]
+      ++ (builtins.concatMap (node: node.dependencies) customNodes)
+    )
+  );
+
+  executable = writers.writeDashBin "comfyui" ''
+    cd $out && \
+    ${pythonEnv}/bin/python comfyui \
+      --input-directory ${inputPath} \
+      --output-directory ${outputPath} \
+      --extra-model-paths-config ${modelPathsFile} \
+      --temp-directory ${tempPath} \
+      "$@"
+  '';
+
+  customNodesCollection = (
+    linkFarm "comfyui-custom-nodes" (
+      builtins.map (pkg: {
+        name = pkg.pname;
+        path = pkg;
+      }) customNodes
+    )
+  );
+in
+stdenv.mkDerivation rec {
+  pname = "comfyui";
+  version = "unstable-2025-04-26";
+
+  src = fetchFromGitHub {
+    owner = "comfyanonymous";
+    repo = "ComfyUI";
+    rev = "b685b8a4e098237919adae580eb29e8d861b738f";
+    hash = "sha256-OtTvyqiz2Ba7HViW2MxC1hFulSWPuQaCADeQflr80Ik=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+    echo "Preparing bin folder"
+    mkdir -p $out/bin/
+    echo "Copying comfyui files"
+    # These copies everything over but test/ci/github directories.  But it's not
+    # very future-proof.  This can lead to errors such as "ModuleNotFoundError:
+    # No module named 'app'" when new directories get added (which has happened
+    # at least once).  Investigate if we can just copy everything.
+    cp -r $src/app $out/
+    cp -r $src/api_server $out/
+    cp -r $src/comfy $out/
+    cp -r $src/comfy_api_nodes $out/
+    cp -r $src/comfy_extras $out/
+    cp -r $src/comfy_execution $out/
+    cp -r $src/utils $out/
+    cp $src/*.py $out/
+    cp $src/requirements.txt $out/
+    mv $out/main.py $out/comfyui
+    echo "Copying ${modelPathsFile} to $out"
+    cp ${modelPathsFile} $out/extra_model_paths.yaml
+    echo "Setting up input and output folders"
+    ln -s ${inputPath} $out/input
+    ln -s ${outputPath} $out/output
+    mkdir -p $out/${tempPath}
+    echo "Setting up custom nodes"
+    ln -snf ${customNodesCollection} $out/custom_nodes
+    echo "Copying executable script"
+    cp ${executable}/bin/comfyui $out/bin/comfyui
+    substituteInPlace $out/bin/comfyui --replace "\$out" "$out"
+    echo "Patching python code..."
+    substituteInPlace $out/folder_paths.py --replace 'os.path.join(base_path, "temp")' '"${tempPath}"'
```

Comment:
SubstituteInPlace can also do multiple replacements in one command

Reply:
Do we really want to do that for some reason? Seems less readable

Reply:
see https://github.com/NixOS/nixpkgs/pull/402112/files#r2085013334 for the example what I had in mind